Bryn Mawr Classical Review.js

```diff
@@ -3,93 +3,135 @@
-		if (prefix == 'x') return ns; else return null;
-	} : null;
-	var arts = new Array();
 	if (detectWeb(doc, url) == "multiple") {
-		var items = new Object();
-		if (doc.evaluate('//table/tbody/tr/td/ul/li/i', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext()) {
-			var boxes = doc.evaluate('//table/tbody/tr/td/ul/li', doc, nsResolver, XPathResult.ANY_TYPE, null);
-			var box;
-			while (box = boxes.iterateNext()) {
-				var link = doc.evaluate('./a', box, nsResolver, XPathResult.ANY_TYPE, null).iterateNext().href;
-				var title = doc.evaluate('./i', box, nsResolver, XPathResult.ANY_TYPE, null).iterateNext().textContent;
-				items[link] = title;
-			}
-		} else if (doc.evaluate('//table/tbody/tr/td/ul/li', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext()) {
-			var title = doc.evaluate('//table/tbody/tr/td/ul/li', doc, nsResolver, XPathResult.ANY_TYPE, null);
-			var next;
-			while (next = title.iterateNext()) {
-				items[next.href]  = Zotero.Utilities.trimInternal(next.textContent);
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
 			}
-		} else if (url.match(/google\.com/)) {
-			var titles = doc.evaluate('//h2[@class="r"]/a[@class="l"]', doc, nsResolver, XPathResult.ANY_TYPE, null);
-			var title;
-			while (title = titles.iterateNext()) {
-				items[title.href] = title.textContent;
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
 			}
-		}
-		items = Zotero.selectItems(items);
-		for (var i in items) {
-			arts.push(i);
-		}
+			ZU.processDocuments(articles, scrape);
+		});
 	} else {
-		arts = [url];
+		scrape(doc, url);
 	}
-	Zotero.Utilities.processDocuments(arts, function(doc) {
-		var item = new Zotero.Item("journalArticle");
-		var title = doc.evaluate('//h3/i', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext().textContent;
-		item.title = "Review of: " + Zotero.Utilities.trimInternal(title);
-		var data = doc.evaluate('//h3[i]', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext().textContent;
-		var title = title.replace("(", "\\(").replace(")", "\\)");
-		var author = doc.evaluate('//b[contains(text(), "Reviewed")]', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext().textContent.match(/Reviewed by\s+([^,]+),/)[1];
-		item.creators.push(Zotero.Utilities.cleanAuthor(author, "author"));
-		var splitRe = new RegExp(title);
-		var authors = data.split(splitRe)[0].replace(/\([^)]+\)/, "").split(/(,|and)\s+/);
-		Zotero.debug(authors);
-		for (var i=0; i<authors.length; i++) {
-			var aut = authors[i];
-			if (aut.match(/\w/) && (aut != "and")) {
-				item.creators.push(Zotero.Utilities.cleanAuthor(aut, "reviewedAuthor"));
-			}
+}
+
+function scrape(doc, url) {
+	var item = new Zotero.Item("journalArticle");
+	
+	var title = ZU.xpathText(doc, '//h3/i');
+	item.title = "Review of: " + Zotero.Utilities.trimInternal(title);
+	var title = title.replace("(", "\\(").replace(")", "\\)");
```

Comment:
this isn't your code, but I find it confusing: it doesn't do anything useful, `title` isn't used again so it's pointless, and `var` isn't needed. If I'm right, delete. Otherwise remove `var`